packages/react/src/components/TextEllipsis/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Ref, useEffect, useState } from 'react';
+import classnames from 'classnames';
+import useSharedRef from '../../utils/useSharedRef';
+import Tooltip, { type TooltipProps } from '../Tooltip';
+
+interface TextEllipsisProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: string;
+  maxLines?: number;
+  as?: React.ElementType;
+  refProp?: string;
+  tooltipProps?: Omit<TooltipProps, 'target' | 'association'>;
+}
```

Comment:
Should we be using the polymorphic type util? https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/utils/polymorphic-type/index.ts

In the docs, you gave an example of specifying `as={Link}`. However, without using the polymorphic type utility, TS would complain if we try to provide properties specific to the component we are rendering `as` (e.g. `href`):
![image](https://github.com/dequelabs/cauldron/assets/6416791/58ae86b0-7856-4527-9747-14a4b396a755)

Reply:
Looks like that polymorphic type doesn't quite pass the mustard here so there's still some work to do. 😞 

<img width="936" alt="" src="https://github.com/dequelabs/cauldron/assets/1062039/26e1a53b-87c8-410f-8edc-87cdeec2b356">

I'm inclined to leave this as-is for now, but I do agree we need to address polymorphic types and likely to address that as a separate issue. What do you think?

Reply:
Digging into this a little bit more, it seems the current `ForwardRefComponent` type only infers types of intrinsic elements. So `as="a"` is able to infer the type, but not `as={Link}`. As such, there's more work to be done here but I would like to defer that to another issue.

Reply:
I'd be fine with deferring it, but I wonder if the `as` prop should be removed or caveats documented, since it's not fully supported.

Reply:
> I'd be fine with deferring it, but I wonder if the as prop should be removed or caveats documented

It _can not_ be removed because the absence of it would create accessibility issues when the element is contained within another interactive element. I've been looking at other instances in different libraries and there doesn't appear to be a way that I've found yet to support non-intrinsic elements.

Reply:
Created https://github.com/dequelabs/cauldron/issues/1407, going to do a bit of digging to see what we need to do here. My guess is if it is possible the type may need to be updated.

Reply:
> It _can not_ be removed because the absence of it would create accessibility issues when the element is contained within another interactive element. I've been looking at other instances in different libraries and there doesn't appear to be a way that I've found yet to support non-intrinsic elements.

I just meant remove it until it's fully supported. But I guess `Link` is probably the worst case scenario. I imagine a lot of button usages would probably work as long as we're just using things like `onClick`